PxWeb/Code/Api2/DataSelection/SelectionHandler.cs

```diff
@@ -115,27 +245,217 @@ private VariablesSelection AddVariables(VariablesSelection variablesSelection, P
         /// </summary>
         /// <param name="variablesSelection"></param>
         /// <returns></returns>
-        private Selection[] MapCustomizedSelection(VariablesSelection variablesSelection)
+        private Selection[] MapCustomizedSelection(PXModel model, VariablesSelection variablesSelection)
         {
             var selections = new List<Selection>();
 
-            foreach (var variable in variablesSelection.Selection)
+            foreach (var varSelection in variablesSelection.Selection)
             {
-                var selection = new Selection(variable.VariableCode);
+                var variable = model.Meta.Variables.GetByCode(varSelection.VariableCode);
+                selections.Add(GetSelection(variable, varSelection));
+            }
+
+            return selections.ToArray();
+        }
+
+        /// <summary>
+        /// Add all values for variable
+        /// </summary>
+        /// <param name="variable">Paxiom variable</param>
+        /// <param name="varSelection">VariableSelection object with wanted values from user</param>
+        /// <returns></returns>
+        private Selection GetSelection(Variable variable, VariableSelection varSelection)
+        {
+            var selection = new Selection(varSelection.VariableCode);
+            var values = new List<string>();
 
-                //Add values if they exist
-                if (!variable.ValueCodes.Count().Equals(0)) 
+            foreach (var value in varSelection.ValueCodes)
+            {
+                if (value.Contains('*'))
                 {
-                    var valueList = variable.ValueCodes[0].ToString().Split(',').ToList();
-                    selection.ValueCodes.AddRange(valueList.ToArray());
+                    AddWildcardStarValues(variable, values, value);
+                }
+                else if (value.Contains('?'))
+                {
+                    AddWildcardQuestionmarkValues(variable, values, value);
+                }
+                else if (value.Contains("TOP("))
```

Comment:
Same case sensitivity and starts with comment as above

Reply:
Now checks that it starts with TOP and is case insensitive.